firmware/ethereum.c

```diff
@@ -157,8 +160,17 @@ static void send_signature(void)
 
 	/* eip-155 replay protection */
 	if (chain_id != 0) {
+		uint8_t data[4];
+		data[0] = (chain_id >> 24) & 0xff;
+		data[1] = (chain_id >> 16) & 0xff;
+		data[2] = (chain_id >> 8) & 0xff;
+		data[3] = (chain_id) & 0xff;
+		int offset = 0;
+		while (!data[offset]) {
```

Comment:
This loop doesn't do bound checking.  It cannot leave the data array, since we know that chain_id is not zero, but I wonder if we should be more defensive here.